### PR TITLE
[BI-768] - Prototype for brapi importer

### PR DESCRIPTION
**Pull request by ctucker3**
The front end for the importer. See bi-api PR for details:

https://github.com/Breeding-Insight/bi-api/pull/80

### CODE REVIEW

File: src/breeding-insight/dao/ImportDAO.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import {ImportData} from "@/breeding-insight/model/import/ImportData";
+import {tblCross} from "@/breeding-insight/dao/mock_data/importMock";
+import {ImportMappingConfig} from "@/breeding-insight/model/import/ImportMapping";
+import * as api from "@/util/api";
+import {BiResponse, Response} from "@/breeding-insight/model/BiResponse";
+
+export class ImportDAO {
+
+  static async getImport(importId: string): Promise<ImportData> {
+    //TODO: Implement once back end is up and running
```
[the remark this answers is not in the document]
Comment (ctucker3) [reply src/breeding-insight/dao/ImportDAO.ts:27]: Not sure, it was likely for the POST/GET pattern of uploading data. Where you POST the data, get an immediate response with an upload id, and then check on the completion status by making a GET call. I can remove this until we implement that.

File: src/views/trait/TraitsImport.vue [hunk 2]
```diff
@@ -48,9 +48,11 @@
     <template v-if="state === ImportState.CHOOSE_FILE || state === ImportState.FILE_CHOSEN">
       <h1 class="title">Import Traits</h1>
       <TraitImportTemplateMessageBox class="mb-5"/>
-      <FileSelectMessageBox v-model="file"
-                               v-bind:fileTypes="'.csv, .xls, .xlsx'"
-                               v-on:import="importService.send(ImportEvent.IMPORT_STARTED)"/>        
+      <div class="box">
```
[the remark this answers is not in the document]
Comment (ctucker3) [reply src/views/trait/TraitsImport.vue:51]: Yep, just removes the box styling from inside the component to the outside so we have a little more control over the look when we reuse it.

File: src/components/import/ColumnSummary.vue [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  @Component({
+    components: {EditIcon}
+  })
+  export default class ColumnSummary extends Vue {
+    @Prop()
+    columns!: string[];
+  }
+</script>
+
+<style scoped>
```
[the remark this answers is not in the document]
Comment (ctucker3) [reply src/components/import/ColumnSummary.vue:49]: Probably just forgot to remove it. I can take that out.

File: src/components/file-import/FileSelectMessageBox.vue [hunk 4]
```diff
@@ -17,7 +17,7 @@
 
 <template>
   <div class="file-select">
-    <div class="box">
+    <div>
```
[the remark this answers is not in the document]
Comment (ctucker3) [reply src/components/file-import/FileSelectMessageBox.vue:20]: See discussion below in the TraitImport file.

File: src/breeding-insight/dao/ImportDAO.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ctucker3) [reply src/breeding-insight/dao/ImportDAO.ts]: pushed

File: src/components/import/ColumnSummary.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (ctucker3) [reply src/components/import/ColumnSummary.vue]: pushed

**Review by nickpalladino — APPROVED** [review]

**Review by timparsons — APPROVED** [review]

File: src/views/import/BrAPIImporter.vue [hunk 5]
```diff
@@ -0,0 +1,761 @@
...
+                v-on:click="selectMapping"
+                class="button is-primary"
+            >
+              Select Import
+            </button>
+          </div>
+        </div>
+      </template>
+      <template v-slot:summary-display>
+        I am a summary
```
[the remark this answers is not in the document]
Comment (ctucker3) [reply src/views/import/BrAPIImporter.vue]: I just remove the text for now so the summary display is empty.

**Pull request merged**